chromedp.go

```diff
@@ -409,6 +411,71 @@ func WithBrowserOption(opts ...BrowserOption) ContextOption {
 	}
 }
 
+// RunResponse is an alternative to Run which can be used with a list of actions
+// that trigger a page navigation, such as clicking on a link or button.
+//
+// RunResponse will run the actions and block until a page loads, returning the
+// HTTP response information for its HTML document. This can be useful to wait
+// for the page to be ready, or to catch 404 status codes, for example.
+//
+// Note that if the actions trigger multiple navigations, only the first is
+// used. And if the actions trigger no navigations at all, RunResponse will
+// block until the context is cancelled.
+func RunResponse(ctx context.Context, actions ...Action) (*network.Response, error) {
+	var resp *network.Response
+	if err := Run(ctx, responseAction(&resp, actions...)); err != nil {
+		return nil, err
+	}
+	return resp, nil
+}
+
+func responseAction(resp **network.Response, actions ...Action) Action {
+	return ActionFunc(func(ctx context.Context) error {
+		respReceived := false
+
+		// First, start listening for events.
+		ch := make(chan struct{})
+		lctx, cancel := context.WithCancel(ctx)
+		ListenTarget(lctx, func(ev interface{}) {
+			switch ev := ev.(type) {
+			case *network.EventResponseReceived:
+				if ev.Type == network.ResourceTypeDocument {
+					respReceived = true
+					if resp != nil {
+						*resp = ev.Response
+					}
+				}
+			case *page.EventLoadEventFired:
+				// Only stop if a load event triggers after we
+				// receive a document response. Otherwise, this
+				// might be a load event from a previous
+				// navigation.
+				if respReceived {
+					cancel()
+					close(ch)
+				}
+			}
+		})
+
+		// Second, run the actions.
+		if err := Run(ctx, actions...); err != nil {
+			return err
+		}
+
+		// Third, block until we have finished loading. If we wanted a
+		// response, check that it isn't nil.
+		select {
+		case <-ch:
+			if resp != nil && *resp == nil {
+				return fmt.Errorf("page loaded without a response")
```

Comment:
nit: without a document type response since we only watch for `ResourceTypeDocument`

Reply:
Good point :)